ReactiveCocoa/Swift/Signal.swift

```diff
@@ -168,25 +156,32 @@ public func take<T, E>(count: Int)(signal: Signal<T, E>) -> Signal<T, E> {
 	precondition(count >= 0)
 
 	return Signal { observer in
+		if count == 0 {
+			sendCompleted(observer)
+			return nil
+		}
+
 		var taken = 0
 
-		return signal.observe(next: { value in
-			if taken < count {
-				taken++
-				sendNext(observer, value)
-			} else {
-				sendCompleted(observer)
+		return signal.observe(Signal.Observer { event in
+			switch event {
+			case let .Next(value):
+				if taken < count {
+					taken++
+					fallthrough
```

Comment:
Definitely confusing at first glance. I also find it a bit disturbing in this context since it's essentially `goto default` from an inner scope.